src/abstractdataframe/abstractdataframe.jl

```diff
@@ -231,42 +232,33 @@ Base.ndims(::AbstractDataFrame) = 2
 Base.similar(df::AbstractDataFrame, dims::Int) =
     DataFrame(Any[similar(x, dims) for x in columns(df)], copy(index(df)))
 
-nas{T}(dv::AbstractArray{T}, dims::@compat(Union{Int, Tuple{Vararg{Int}}})) =   # TODO move to datavector.jl?
-    DataArray(Array(T, dims), trues(dims))
-
-nas{T,R}(dv::PooledDataArray{T,R}, dims::@compat(Union{Int, Tuple{Vararg{Int}}})) =
-    PooledDataArray(DataArrays.RefArray(zeros(R, dims)), dv.pool)
-
-nas(df::AbstractDataFrame, dims::Int) =
-    DataFrame(Any[nas(x, dims) for x in columns(df)], copy(index(df)))
-
 ##############################################################################
 ##
 ## Equality
 ##
 ##############################################################################
 
-function Base.isequal(df1::AbstractDataFrame, df2::AbstractDataFrame)
-    size(df1, 2) == size(df2, 2) || return false
-    isequal(index(df1), index(df2)) || return false
+# Imported in DataFrames.jl for compatibility across Julia 0.4 and 0.5
+function @compat(Base.:(==))(df1::AbstractDataFrame, df2::AbstractDataFrame)
```

Comment:
Yeah, but let's continue discussing this at https://github.com/JuliaStats/NullableArrays.jl/issues/84 to avoid making this PR too messy. I guess I'll switch it to returning a `Bool` since it seems that's the general preference.